### ISSUE TITLE
Unable to save attribute table after editing in QGIS

### ISSUE THREAD

**Issue by jbelleramos** [qgis/QGIS#56148]
### What is the bug or the crash?

I couldn't save an  attribute table as it displays an error.

I have a table layer on QGIS with bunch of data inside. I opened its attribut table to edit some stuff, but when I try to save after editing, it would display an error saying "_OGR error syncing to disk: Cannot rename files_" or "_Commit errors: Could not commit changes to layer_".

I downloaded the OSGeo4W installer and installed the 3.28.12 Firenze LTR version but it still does not work.

### Steps to reproduce the issue

![qgis error](https://github.com/qgis/QGIS/assets/158568267/9bde835c-94d5-4550-a36d-e1a5413a11ff)

### Versions

QGIS version
3.28.13-Firenze
QGIS code revision
3b4db4f561
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.8.0
PROJ version
9.3.0
EPSG Registry database version
v10.094 (2023-08-08)
GEOS version
3.12.1-CAPI-1.18.1
SQLite version
3.41.1
PDAL version
2.6.0
PostgreSQL client version
unknown
SpatiaLite version
5.1.0
QWT version
6.1.6
QScintilla2 version
2.13.4
OS version
Windows 10 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by roya0045** [comment]
What is the file type?

**Comment by ERHash** [comment]
The file is in local disk ? Not open in other app ?

**Comment by jbelleramos** [comment]
> What is the file type?

The file type is in CSV. So here's the thing, I just downloaded the QGIS project file I was supposed to work on because it was prepared by a different person. When I opened the QGIS project file on my QGIS app, it didn't automatically load all the CSV file that's supposed to go with the layers. So I manually added the CSV files to edit the attribute table, and then that issue that it wouldn't save the edits I made on the table.

Both the QGIS Project file & the csv files are inside the same folder on a local disk.
 
![qgis ss](https://github.com/qgis/QGIS/assets/158568267/5fdea4c4-8889-478f-825d-9582f0971ecf)

**Comment by jbelleramos** [comment]
> The file is in local disk ? Not open in other app ?

Yes, both the QGIS Project file and the CSV file for the table are in the same folder in local disk.

So here's the thing, I just downloaded the QGIS project file I was supposed to work on because it was prepared by a different person. When I opened the QGIS project file on my QGIS app, it didn't automatically load all the CSV file that's supposed to go with the layers. So I manually added the CSV files to edit the attribute table, and then that issue that it wouldn't save the edits I made on the table.
